### PR TITLE
CLDR-13731 Report in ConsoleCheck if alt without non-alt

### PR DESCRIPTION
**Pull request by btangmu**
-New CheckAltOnly.java to check if only the alt path is present but the non-alt path is absent

-New CheckCLDR.CheckStatus.Subtype.missingNonAltPath

-New TestCheckAltOnly.java (unit test for CheckAltOnly.java)

-CheckAltOnly: no error for the root locale

-CheckAltOnly: no error if the non-alt path has a constructed value

-CheckAltOnly: no error if getSourceLocaleIdExtended does not return XMLSource.CODE_FALLBACK_ID

-CheckAltOnly: no error if currency symbol with alt is same as root alt value

-XML data changes: remove alt from error-triggering paths (move to non-alt) for locale display names

-XML data changes: delete currency types RUR in pa.xml and GEL in to.xml

-Fix TestCheckDisplayCollisions: frCaResolved was clearly intended to be resolved

-Fix some warnings

-Remove some dead/commented-out code

-Change some public to private

-Comments

CLDR-13731

- [x] This PR completes the ticket.

### CODE REVIEW

**Review by macchiati — APPROVED** [review]

**Comment by btangmu** [comment]
@macchiati @srl295 now this is approved, it still has 51 failing data tests as expected, so two options are:

1. fixing the data before merging to main
2. merging to main in spite of the failures (overriding github required status -- how?), then fixing the failures in a subsequent PR

I imagine pros and cons both ways. Would you like me to try to fix the data (with some guidance)? If so, then maybe I should simply combine the code and data changes in a subsequent PR. On the other hand, if others than me are going to fix the data, it could be convenient to have the tests committed now -- though also maybe a nuisance to have ongoing errors on main

**Comment by btangmu** [comment]
BTW about overriding github required status, I've seen these:

PR Flags

The following "variables" can be set in the pull request body to change the behavior of the checker.

// Allow commits with mixed issue tags
DISABLE_JIRA_ISSUE_MATCH=true

// Allow multiple commits in the same PR
ALLOW_MANY_COMMITS=true

https://github.com/unicode-org/jira-github-pr-check

... but I don't see a way to override cldr-mvn/check errors...

**Comment by macchiati** [comment]
There is still something a odd with the test.

*2021-10-26T16:51:12.4860622Z ff_Adlm [Fulah (Adlam)]*

▸Currencies|Western_Asia_(C)|Western_Asia:_Lebanon|LBP-symbol-narrow◂

〈L£〉

〈L£〉

⁅missing non alt path⁆

❮Error: A path with alt is present but the corresponding path without alt
is missing; a solution might be to confirm the non-alt path❯

*https://st.unicode.org/cldr-apps/v#/ff_Adlm//714ef3a5e09c1bb3*
<https://st.unicode.org/cldr-apps/v#/ff_Adlm//714ef3a5e09c1bb3>

root

The table indicates (and clicking on the link) that there is a value in
root. https://st.unicode.org/cldr-apps/v#/root/C_WAsia/7a65298807a5c2e0
Now, what may be happening is that in the ST we are reporting as "root"
items that are also just in code-fallback. Can you confirm that this is the
case?

As to the question, we should be able to fix these as part of the checkin.
Please create a spreadsheet to copy in the table, and we can review and
indicate the right change. Key questions.

   1. If the alt path is not in root, we want to know about it. In case it
   was mistakenly entered with a bulk upload, we may want to delete the alt
   path as the solution.
   2. For the others, we will generally want to add the explicit non-alt
   value


Note: when we finish the release, we also need to reset the Phase for
checking, so that introducing errors is allowed. Maybe we don't make that
clear enough though.

(When we add a test that exposes an error, we generally want to allow for
the vetters to fix it because we don't know how to. In this case, it is
much easier.)

Mark

**Comment by srl295** [comment]
I missed the discussion around over writing checks, but I really don’t
think we should do that. If we can avoid it. There’s actually a way to
create a Q of requests to be merged and then another pull request could be
added to fix the errors if that’s what’s needed

**Comment by macchiati** [comment]
We have often fixed data in the same PR that we fixed (or added to) a test.
Is it the volume of changes that concerns you?


Mark

**Comment by srl295** [comment]
No, I think it should be fixed in the same pull request, I thought there
was a discussion about merging this to main with errors.

**Comment by btangmu** [comment]
@macchiati @srl295 my understanding is that both of you prefer my option "1. fixing the data before merging to main"

in that case, the question of how to override is moot (for now at least) since that would have been my option "2."

next I should address Mark's question about something "...odd with the test..."

**Comment by btangmu** [comment]
@macchiati I'm attaching the table (as an xlsx file) here:
[51.xlsx](https://github.com/unicode-org/cldr/files/7455293/51.xlsx)

**Comment by btangmu** [comment]
@macchiati @srl295 a simple and obvious fix would be to remove from the xml the values for the alt paths in question. I imagine that vetters could subsequently import their old votes for those alt values, and vote for them again along with voting for corresponding non-alt values -- is that right? Is there a better way?

**Comment by srl295** [comment]
> Now, what may be happening is that in the ST we are reporting as "root"
> items that are also just in code-fallback. Can you confirm that this is the
> case?

That's not what's happening here. This data is in `root.xml` and that is the cause of at least some of these.

```xml
                        <currency type="PKR">
                                <symbol alt="narrow">Rs</symbol>
                        </currency>
                        <currency type="PLN">
                                <symbol alt="narrow">zł</symbol>
                        </currency>
                        <currency type="PYG">
                                <symbol alt="narrow">₲</symbol>
                        </currency>
                        <currency type="RON">
                                <symbol alt="narrow">lei</symbol>
                        </currency>
                        <currency type="RUB">
                                <symbol alt="narrow">₽</symbol>
                        </currency>
                        <currency type="RWF">
                                <symbol alt="narrow">RF</symbol>
                        </currency>
                        <currency type="SBD">
                                <symbol alt="narrow">$</symbol>
                        </currency>
                        <currency type="SEK">
                                <symbol alt="narrow">kr</symbol>
                        </currency>
                        <currency type="SGD">
                                <symbol alt="narrow">$</symbol>
                        </currency>
                        <currency type="SHP">
                                <symbol alt="narrow">£</symbol>
                        </currency>
                        <currency type="SRD">
                                <symbol alt="narrow">$</symbol>
                        </currency>
                        <currency type="SSP">
                                <symbol alt="narrow">£</symbol>
                        </currency>
                        <currency type="STN">
                                <symbol alt="narrow">Db</symbol>
                        </currency>
                        <currency type="SYP">
                                <symbol alt="narrow">£</symbol>
                        </currency>
                        <currency type="THB">
                                <symbol alt="narrow">฿</symbol>
                        </currency>
                        <currency type="TOP">
                                <symbol alt="narrow">T$</symbol>
                        </currency>
```

Should we exclude currency symbols here? Because I think this is a case where the normal symbol would be, say, `THB` even if the narrow is `฿`.

Alternatively (if you pardon the pun!) perhaps _for completeness_ we should explicitly update all of these to be:

```xml
                        <currency type="RUB">
                                <symbol>RUB</symbol>  <!-- ««« Add these (from code-fallback) explicitly -->
                                <symbol alt="narrow">₽</symbol>
                        </currency>
```

**Comment by srl295** [comment]
- `yi` name for `TL` - this should _move_ from TL-variant to TL. For some reason they translated the variant but not the regular one.  This seems like exactly what the test is designed to catch.

**Comment by srl295** [comment]
These other three are also good positive fails.

- `ig` Kurdish - just move these from alt to regular. Another good positive, the translator should translate the non-alt first. This one can be solved with a move, I think.

- All of the ones around MO, HK, CD, CG:  The translators need to complete the work here: in most cases they've *declined to translate the parenthetical* SAR, SAR, Kinshasa, Brazzaville,  respectively. Here I agree to just delete these entries so they can be sorted out.
- `CZ` and `PS` corsican translator declined to translate "Czechia" and "Palestinian Territories". Delete the alt form, redo both. Another good hit.

**Comment by srl295** [comment]
I think I've reviewed all of the data errors in the spreadsheet, with two possible solutions for the currency part.

**Comment by btangmu** [comment]
@srl wrote "This data is in root.xml and that is the cause of at least some of these ..."
The examples you gave in root are all "alt" paths. However, CheckAltOnly is interested in the source locales of the "non-alt" paths (corresponding to "alt" paths in non-root locales). So, it's not clear to me whether these examples are relevant for this ticket -- if they are, what's a particular example out of the 51 errors?

**Comment by macchiati** [comment]
Mark


On Mon, Nov 1, 2021 at 1:31 PM Tom Bishop ***@***.***> wrote:

> @macchiati <https://github.com/macchiati> Yes I can confirm that in the
> ST we are reporting as "root" items that are also in code-fallback. The
> front end, by design, changes "code-fallback" to "root" in such links per
> an old completed ticket (
> https://unicode-org.atlassian.net/browse/CLDR-11622 ).
>
That's what I thought might be happening. What is a mystery to me is why
other cases like STN don't also show up as 'root'. They should also be
coming from code-fallback.

For the non-alt path in question in root (
> http://localhost:9080/cldr-apps/v#/root/C_WAsia/7a65298807a5c2e0 ), the
> Info Panel says for "LBP", "This item is an untranslated code" and "This
> item is constructed from other values". The status in root is "Missing".
> The test (CheckAltOnly) would not trigger an error unless the source locale
> were "code-fallback" (on the back end).
>
OK, glad you confirmed.

> BTW the message "This item is constructed from other values" is
> potentially confusing: it's actually used for "code-fallback" in general,
> not to be confused with CLDRFile.getConstructedValue which returns null
> except for a very limited set of paths that always contain
> "//ldml/localeDisplayNames/languages/language".
>
Hmmm. We probably should change that message.

**Comment by macchiati** [comment]
Yes, that is probably an error.

**Comment by btangmu** [comment]
@macchiati @srl295 here's a google sheet with the 51 errors:
https://docs.google.com/spreadsheets/d/1rXa1XXyP76hrspLtHSis4uH6FJqzAUzcwVNoOYmMUow/edit#gid=1818827278
I've also linked to it from today's infrastructure agenda

**Comment by macchiati** [comment]
Thanks!

Mark

**Comment by btangmu** [comment]
@macchiati @srl295 there are still two console-check errors, both currency symbols with alt, but NOT same as root alt value:

https://st.unicode.org/cldr-apps/v#/pa//386035af20c8d53a

https://st.unicode.org/cldr-apps/v#/to//268bf25be49df55b

**Comment by srl295** [comment]
I think they are both leftover from old root values and should be deleted.

One is the OLD Russian ruble as `p.` and the other is the georgian Lek symbol when it was a variant and not in the main narrow.

File: tools/cldr-code/src/test/java/org/unicode/cldr/unittest/TestCheckDisplayCollisions.java
Comment (srl295) [inline tools/cldr-code/src/test/java/org/unicode/cldr/unittest/TestCheckDisplayCollisions.java:86]: 👍

**Review by srl295 — APPROVED** [review]

**Comment by srl295** [comment]
i approve pending test fixes of course…

**Comment by btangmu** [comment]
> I think they are both leftover from old root values and should be deleted.
> 
> One is the OLD Russian ruble as `p.` and the other is the georgian Lek symbol when it was a variant and not in the main narrow.

I'll delete both and update this PR

**Review by srl295 — APPROVED** [review]

**Pull request merged**
